engine/shard.go

```diff
@@ -1843,16 +1841,12 @@ func (s *shard) GetRPName() string {
 }
 
 func (s *shard) DisableDownSample() {
-	if atomic.CompareAndSwapInt32(&s.downSampleEn, 1, 0) {
-		close(s.stopDownSample)
-	}
+	s.stopDownSample.Close()
```

Comment:
I don't know what causes the panic in the original code , but I think it is doube close of the channel.
if this is the case, then I think the next statement `s.dswg.Wait()` may not return for a long time in the below case:

1. the 1st goroutine calls `DisableDownSample()` (assume the internal counter of `s.stopDownSample` becomes 1 and then suspends because of `s.dswg.Wait()`
2. the 2nd goroutine calls `EnableDownSample()` (the internal counter of `s.stopDownSample becomes 0) and then calls `s.dswg.Add(1)`
3. the 1st goroutine can not continue before the 2nd goroutine calls 's.dswg.Done()`.

I'm not familiar enough with code, that's I'm not sure if the above case could happen, but I think it is very possible, so propose make a check.

Reply:
这是一个好问题，我单独提了一个 issues 可以在这里讨论

https://github.com/openGemini/openGemini/issues/511
@localvar